README.md

```diff
@@ -9,3 +9,664 @@ ExpressRouterAdapter is available as a package on NPM for use with a module bund
+
+## What if I need a different handler based on the media type?
+TODO
+
+## What if I need to lock down most of my endpoints?
+So far, we haven't done any authorization, lets lock it down.
+
+If we remove allow anonymous, we will start getting a 401
+```ts
+new RouterMetaBuilder()
+  .path('/pets/')
+  .mediaType(this.petsMediaType)
+  .get(() => {
+    return this.pets;
+  }),
+```
+
+```sh
+curl -H 'accept:application/pets+json'  http://localhost:4000/pets/
+{"status":401,"message":"Authorization is required"}
+```
+
+Security is going to be different for every application, so we provide a hook for you to bring your own security.
+
+
+```ts
+
+import 'reflect-metadata'; // Required by aurelia-dependency-injection
+
+import { ISecurityContext, ISecurityContextProvider, RouteProvider, SecurityContextProvider } from '@symbiotic/express-router-adapter';
+import { Container } from 'aurelia-dependency-injection';
+import { ApplicationRouteProvider } from './ApplicationRouteProvider';
+
+class ApplicationSecurityContext implements ISecurityContext {
+  constructor(public principal: string) { }
+  toLogSafeString(): string {
+    return this.principal;
+  }
+}
+// tslint:disable-next-line: max-classes-per-file
+class ApplicationSecurityContextProvider implements ISecurityContextProvider {
+  async getSecurityContext({ req }: any): Promise<ApplicationSecurityContext> {
+    console.log('in here?');
+    const authHeader = req.headers.authorization;
+    // this is NOT real security
+    return new ApplicationSecurityContext(authHeader);
+  }
+
+}
+
+export const getContainer = async (): Promise<Container> => {
+  const container = new Container();
+
+  container.registerAlias(ApplicationRouteProvider, RouteProvider);
+  container.registerAlias(ApplicationSecurityContextProvider, SecurityContextProvider);
+
+  return container;
+};
+```
+
+Now we get the following
+```sh
+curl -H 'authorization:jon' -H 'accept:application/pets+json'  http://localhost:4000/pets/
+[{"firstName":"honey","lastName":"hoguet"},{"firstName":"poseidon","lastName":"hoguet"}]
+```
+
+Which of course isn't secure! But it demonstrates the plumbing.
+
+> Of note: you must have a principal on your security context to not be considered anonymous. You will still get a 401 without a principal property. This should be reconsidered in the next version.
+
+## What if I want a base path added?
+TODO
+
+## What if I need custom timeouts?
+TODO
+
+## How can I provide my own logger?
+TODO
+
+## How do I serialize a response when it isn't json
+TODO
+
+
+## Motivation
+ExpressRouterAdapter was born out of necessity as Symbiotic Labs supported multiple clients building out Restful APIs using Node.
+
+We wanted some basics of REST to be easy / out of the box, so that developers can focus on using REST.
+ - You should just be able to return a model and get a JSON response. You shouldn't have to think about 200, or JSON formatting it.
+ - You should not have to return anything and get 204.
```

Comment:
Consider rephrasing, 'You should be able to produce a 204 by NOT returning a value from your handler'